types.go

```diff
@@ -43,6 +46,22 @@ func (c *common) GetFullName() string { return c.FullName }
 // IsProto3 returns whether or not this is a proto3 object
 func (c *common) IsProto3() bool { return c.file.GetSyntax() == "proto3" }
 
+func (c *common) setOptions(options proto.Message) {
+	for _, extension := range proto.RegisteredExtensions(options) {
+		if !proto.HasExtension(options, extension) {
+			continue
+		}
+		ext, err := proto.GetExtension(options, extension)
+		if err != nil {
+			continue
+		}
+		if c.OptionExtensions == nil {
+			c.OptionExtensions = make(map[string]interface{})
```

Comment:
Rather than doing this here, can we make `newCommon` initialize the map? That way downstream apps don't need to check for `nil` before accessing a value

Reply:
Accessing (reading) fields in a `nil` map has the same results as accessing them from an empty map. The only problem would be with writing, but that shouldn't happen downstream.